File: python/sum-of-multiples/sum_of_multiples.py
def sum_of_multiples(limit, factors):
	multiples = filter(lambda x: is_multiple(x, factors), range(1,limit))
	return reduce((lambda x, y: x + y), multiples, 0)

def is_multiple(n, factors):
	for f in factors:
		if f != 0:
			if n % f == 0:
				return True
	return False

File: ruby/sum-of-multiples/sum_of_multiples.rb
class SumOfMultiples
  def initialize *input
    @factors = input
  end

  def to limit
    (0...limit).to_a.keep_if { |n| is_multiple n }.reduce(:+)
  end

  def is_multiple n
    @factors.each do |f|
      return true if n%f == 0
    end
    false
  end
end